buildAddOn.py

```diff
@@ -0,0 +1,253 @@
+    addOnName = configData['addOnName']
+    optionalParams = None
+    if 'addOnSpecificCMakeParameters' in configData:
+        optionalParams = configData['addOnSpecificCMakeParameters']
+
+    buildPath = buildFolder / addOnName / version
+    if languageCode is not None:
+        buildPath = buildPath / languageCode
+
+    # Add params to configure cmake
+    projGenParams = []
+    projGenParams.append ('cmake')
+    projGenParams.extend (['-B', str(buildPath)])
+
+    if platformName == 'WIN':
+        projGenParams.append (f'-G {configData["winCMakeProjectGenerator"]}')
+        projGenParams.append (f'-T {configData["winCMakePlatformToolset"]}')
+    elif platformName == 'MAC':
+        projGenParams.extend (['-G', 'Xcode'])
+
+    projGenParams.append (f'-DAC_API_DEVKIT_DIR={str(devKitFolder / "Support")}')
+    projGenParams.append (f'-DCMAKE_BUILD_TYPE={configuration}')
+
+    if languageCode is not None:
+        projGenParams.append (f'-DAC_ADDON_LANGUAGE={languageCode}')
+
+    if optionalParams is not None:
+        for key in optionalParams:
+            projGenParams.append (f'-D{key}={optionalParams[key]}')
+
+    projGenParams.append (str(rootFolder))
+
+    projGenResult = subprocess.call (projGenParams)
+    if projGenResult != 0:
+        print ('Failed to generate project')
+        return 1
+
+    # Add params to build AddOn
+    buildParams = []
+    buildParams.append ('cmake')
+    buildParams.extend (['--build', str(buildPath)])
+    buildParams.extend (['--config', configuration])
+
+    buildResult = subprocess.call (buildParams)
+    if buildResult != 0:
+        print ('Failed to build project')
+        return 1
+
+    return 0
+
+def CopyResultToPackage(packageRootFolder, buildFolder, version, addOnName, platformName, configuration, languageCode=None, isRelease=False):
+    packageFolder = packageRootFolder / version
+    sourceFolder = buildFolder / addOnName / version
+
+    if languageCode is not None:
+        packageFolder = packageFolder / languageCode
+        sourceFolder = sourceFolder / languageCode
+    sourceFolder = sourceFolder / configuration
+
+    if not packageFolder.exists():
+        packageFolder.mkdir(parents=True)
+
+    fileName = addOnName
+    if not isRelease:
+        fileName = f'{fileName}_{configuration}'
+
+    if platformName == 'WIN':
+        shutil.copy (
+            sourceFolder / f'{addOnName}.apx',
+            packageFolder / f'{fileName}.apx',
+        )
+        if configuration == 'Debug':
+            shutil.copy (
+            sourceFolder / f'{addOnName}.pdb',
+            packageFolder / f'{fileName}.pdb',
+        )
+
+    elif platformName == 'MAC':
+        subprocess.call ([
+            'cp', '-r',
+            sourceFolder / f'{addOnName}.bundle',
+            packageFolder / f'{fileName}.bundle'
+        ])
+
+
+
+def Main():
+    args = ParseArguments()

```

Comment:
Let divide the Main function into different sections:
- ParseArguments
- PrepareParameters
- BuildAddOns
- PackageAddOns
Separate them by moving the relevant code sections into functions.

Reply:
Done